projects/distributed-tracing/src/shared/dashboard/widgets/trace-detail/data/trace-detail-data-source.model.ts

```diff
@@ -43,7 +43,7 @@ export class TraceDetailDataSourceModel extends GraphQlDataSourceModel<TraceDeta
   }
 
   protected getTraceAttributes(): string[] {
-    return ['tags', 'statusCode', ...this.attributes];
+    return ['tags', 'statusCode', 'traceId', 'apiTraceId', ...this.attributes];
```

Comment:
So along the lines of my comment on waterfall, this is requesting attributes now that only exist on api traces - a regular trace would error. If this data source is only used for api traces (not sure, but the ambiguity and reuse of code between trace types has caused lots of headaches), then we should rename it as such - else, we either need to fork based on trace type or revert.

Reply:
So, we can get rid of apiTraceId here, but traceId we still need since it is the correct and actual Trace ID.

Reply:
Looking some more, I'm 99% sure that all of these are API_TRACE calls. The only place I see the full trace call is from the trace-waterfall-data-source.model. But you are right, this is a bit of a mess and I'm not 100%.

I'm going to rename these to api- to make them more specific.

Reply:
Tested this change in waterfall, full waterfall, and all of the trace tabs (and their links to waterfall). Seems to be fine.